hive/replays/ppo_replay.py

```diff
@@ -0,0 +1,163 @@
+            stack_size (int): The number of frames to stack to create an observation.
+            n_step (int): Horizon used to compute n-step return reward
+            gamma (float): Discounting factor used to compute n-step return reward
+            use_gae (bool): Whether to use generalised advantage estimates for calculating
+                returns
+            gae_lambda (float): Discouting factor used to compute generalised advantage
+                estimation
+            observation_shape: Shape of observations that will be stored in the buffer.
+            observation_dtype: Type of observations that will be stored in the buffer.
+                This can either be the type itself or string representation of the
+                type. The type can be either a native python type or a numpy type. If
+                a numpy type, a string of the form np.uint8 or numpy.uint8 is
+                acceptable.
+            action_shape: Shape of actions that will be stored in the buffer.
+            action_dtype: Type of actions that will be stored in the buffer. Format is
+                described in the description of observation_dtype.
+            action_shape: Shape of actions that will be stored in the buffer.
+            action_dtype: Type of actions that will be stored in the buffer. Format is
+                described in the description of observation_dtype.
+            reward_shape: Shape of rewards that will be stored in the buffer.
+            reward_dtype: Type of rewards that will be stored in the buffer. Format is
+                described in the description of observation_dtype.
+            extra_storage_types (dict): A dictionary describing extra items to store
+                in the buffer. The mapping should be from the name of the item to a
+                (type, shape) tuple.
+            num_players_sharing_buffer (int): Number of agents that share their
+                buffers. It is used for self-play.
+        """
+        if extra_storage_types is None:
+            extra_storage_types = dict()
+        extra_storage_types.update(
+            {
+                "values": (np.float32, ()),
+                "returns": (np.float32, ()),
+                "advantages": (np.float32, ()),
+                "logprob": (np.float32, ()),
+            }
+        )
+        super().__init__(
+            capacity + stack_size - 1,
+            stack_size,
+            n_step,
+            gamma,
+            observation_shape,
+            observation_dtype,
+            action_shape,
+            action_dtype,
+            reward_shape,
+            reward_dtype,
+            extra_storage_types,
+            num_players_sharing_buffer,
+        )
+        self._use_gae = use_gae
+        self._gae_lambda = gae_lambda
+
+    # Taken from https://github.com/vwxyzjn/ppo-implementation-details/blob/main/ppo_shared.py
+    def compute_advantages(self, values):
```

Comment:
Instead of this, can you maybe just have the buffer take in a (registered) function as an argument that will then compute the advantages? That way, if someone wants to compute advantages in a different way, they can use the same replay, just pass a different function for computing the advantages. If you do this, you should have this function just take in the rewards and values and return advantages/returns. 
I am also not sure why this computation needs to be done in the replay buffer? Can you not just do it in the agent, and then use the indices returned in the batch. Not sure either way, but that might be a better division of responsibilities. Let me know if you have an argument for the other way.